src/qtpromise/qpromise_p.h

```diff
@@ -553,6 +556,76 @@ struct PromiseInspect
     }
 };
 
+template<typename T, typename U, typename Enable = void>
+struct PromiseConverter;
+
+template<typename T, typename U>
+struct PromiseConverter<
+    T,
+    U,
+    typename std::enable_if<!std::is_same<T, void>::value && !std::is_same<U, void>::value
+                                && !std::is_same<T, QVariant>::value
+                                && !std::is_same<U, QVariant>::value,
+                            void>::type>
+{
+    static std::function<U(const T&)> create()
+    {
+        return [](const T& value) {
+            return static_cast<U>(value);
+        };
+    }
+};
+
+template<typename T>
+struct PromiseConverter<T, void>
```

Comment:
True, [1] says that `static_cast<void>` discards the expression after evaluating it. Since we don't really have an expression with side effects, I've removed the U=void specialization

[1] https://en.cppreference.com/w/cpp/language/static_cast